src/app/menu/config/config.component.ts

```diff
@@ -1,13 +1,22 @@
 import { Component, OnInit } from '@angular/core';
+import { Account } from '../../models/account.model';
+import { NgxSmartModalService } from 'ngx-smart-modal';
 
 @Component({
   selector: 'app-config',
   templateUrl: './config.component.html',
   styleUrls: ['./config.component.scss']
 })
 export class ConfigComponent implements OnInit {
-
-  constructor() { }
+  public account: Account;
+  public accounts: Account[] = [];
+  
+  AddAccount(accountName: string){
+    this.account= new Account(accountName);
+    this.accounts.push(this.account);
+  }
```

Comment:
Para optimizarlo:
this.account.push(new Account(accountName));
y borrar propiedad account porque no te sirve para nada